### PR TITLE
Apply correct SELinux label to kubelet volume dir in local cluster

### PR DESCRIPTION
**Pull request by aveshagarwal**
This commit applies correct selinux label to /var/lib/kubelet on selinux enabled systems when kube cluster is started with hack/local-up-cluster.sh. Otherwise various e2e test cases  (PreStop, ServiceAccounts, Secrets etcd) fail with error caused due to selinux avcs. For example PreStop e2e's webserver container failed with 

Unable to create client; error: open /var/run/secrets/kubernetes.io/serviceaccount/token: permission denied  

Due to following selinux avcs:
type=AVC msg=audit(1446050473.258:7210): avc: denied { read } for pid=9992 comm="webserver" name="token" dev="tmpfs" ino=499599 scontext=system_u:system_r:svirt_lxc_net_t:s0:c532,c940 tcontext=unconfined_u:object_r:var_lib_t:s0 tclass=file permissive=0

### CODE REVIEW

**Comment by aveshagarwal** [comment]
@pmorie

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -253,7 +253,22 @@ function start_controller_manager {
 
 function start_kubelet {
     KUBELET_LOG=/tmp/kubelet.log
+
     if [[ -z "${DOCKERIZE_KUBELET}" ]]; then
+      mkdir -p /var/lib/kubelet
```
Comment (pmorie) [inline hack/local-up-cluster.sh]: should be outside this block

Comment (aveshagarwal) [inline hack/local-up-cluster.sh]: At first I just thought not to affect else block with this change.

Comment (pmorie) [inline hack/local-up-cluster.sh]: Actually...

You can keep this here, and then on line 295: change this:

```
 --volume=/var/lib/kubelet/:/var/lib/kubelet:rw \
```

to:

` --volume=/var/lib/kubelet/:/var/lib/kubelet:rwZ \`

...and docker will relabel the volume if selinux is enabled in docker.

Comment (aveshagarwal) [inline hack/local-up-cluster.sh]: @pmorie, fixed that and pushed updated PR.

**Comment by pmorie** [comment]
@aveshagarwal Did you test with containerized kubelet and without?  Do the E2Es pass against both clusters?

**Comment by aveshagarwal** [comment]
@pmorie I have only tested with non-containerized kubelet as that is the default in hack/local-up-cluster.sh

**Comment by pmorie** [comment]
@aveshagarwal give it a shot with containerized as well

**Comment by aveshagarwal** [comment]
@pmorie ok will do soon.

**Comment by aveshagarwal** [comment]
@pmorie pushed updated PR after fixing it for dockerized kubelet. Please note that this indeed solves selinux issue. Also, I had to move "mkdir -p /var/lib/kubelet" outside if block because otherwise in else block, it erred that the dir does not exist.

**Comment by aveshagarwal** [comment]
@pmorie all checks passed.

**Comment by aveshagarwal** [comment]
@pmorie Tested it with DOCKERIZE_KUBELT and e2e for secrets passed. However for emptydir, out of 18, 14 passed and 4 failed.

### Secrets e2e test:

`e2e.test --host="127.0.0.1:8080" --provider="local" --ginkgo.v=true -ginkgo.focus="Secrets" --kubeconfig="/root/.kube/config"`

```
• [SLOW TEST:27.088 seconds]
Secrets
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/secrets.go:100
  should be consumable from pods [Conformance]
  /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/secrets.go:99
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 1 of 196 Specs in 27.097 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 195 Skipped PASS
````

### Emptydir  e2e tests (18 tests):
`e2e.test --host="127.0.0.1:8080" --provider="local" --ginkgo.v=true -ginkgo.focus="EmptyDir" --kubeconfig="/root/.kube/config"`

```
Summarizing 4 Failures:

[Fail] EmptyDir volumes [It] volume on tmpfs should have the correct mode using FSGroup [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is non-root [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] volume on default medium should have the correct mode using FSGroup [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is root [Conformance] [Skipped] 
/root/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/util.go:1118

Ran 18 of 196 Specs in 313.155 seconds
FAIL! -- 14 Passed | 4 Failed | 0 Pending | 178 Skipped --- FAIL: TestE2E (313.16s)
FAIL
```

**Comment by aveshagarwal** [comment]
@pmorie But I did not see any new avcs so my guess is that the e2e test failures are not related to selinux. So this PR seems to be working fine.

**Comment by pmorie** [comment]
@aveshagarwal you need to use `ALLOW_SECURITY_CONTEXT=1` when using `local-up-cluster.sh` for those tests to pass.

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -253,7 +253,22 @@ function start_controller_manager {
 
 function start_kubelet {
     KUBELET_LOG=/tmp/kubelet.log
+
+    mkdir -p /var/lib/kubelet
     if [[ -z "${DOCKERIZE_KUBELET}" ]]; then
+      # On selinux enabled systems, it might
+      # require to relabel /var/lib/kubelet
+      if which selinuxenabled &>/dev/null && \
```
Comment (pmorie) [inline hack/local-up-cluster.sh]: nit: `>& /dev/null` with a space between redirect and `/dev/null`

**Comment by aveshagarwal** [comment]
@pmorie, fixed and pushed updated PR.

**Comment by pmorie** [comment]
@aveshagarwal LGTM, thanks for the fix.

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**